python/ngen_conf/src/ngen/config/path_pair/_abc_mixins.py

```diff
@@ -0,0 +1,74 @@
+from abc import ABC, abstractmethod
+
+from typing import Generic, Iterable, Optional
+from typing_extensions import Self
+
+from .typing import StrPath, T
+
+
+class AbstractPathPairMixin(ABC, Generic[T]):
```

Comment:
Would like to see some inline docstrings describing the intended use/semantics of the abstract class and methods